src/rdkafka_fetcher.c

```diff
@@ -655,12 +789,21 @@ rd_kafka_fetch_reply_handle(rd_kafka_broker_t *rkb,
                         rkt = rd_kafka_topic_find0(rkb->rkb_rk, &topic);
                 }
 
+                fetch_tags.Topics[i].TopicId = topic_id;
+
                 rd_kafka_buf_read_arraycnt(rkbuf, &PartitionArrayCnt,
                                            RD_KAFKAP_PARTITIONS_MAX);
```

Comment:
Same conditional allocation
```suggestion
                rd_kafka_buf_read_arraycnt(rkbuf, &PartitionArrayCnt,
                                           RD_KAFKAP_PARTITIONS_MAX);
                if (rd_kafka_buf_ApiVersion(request) >= 12) {
                        rd_kafkap_Fetch_reply_tags_set_topic(
                            &FetchTags, i, topic_id, PartitionArrayCnt);
                }
```

Reply:
Updated